pkg/watcher/watcher.go

```diff
@@ -0,0 +1,82 @@
+package watcher
+
+import (
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/fsnotify/fsnotify"
+
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/logger"
+)
+
+// WatchForUpdates performs an action every time a file on disk is updated
+func WatchFileForUpdates(filename string, done <-chan bool, action func()) {
+	filename = filepath.Clean(filename)
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		logger.Fatalf("failed to create watcher for '%s': %s", filename, err)
```

Comment:
Similar comment to before, I think this is a bad practice to not return an error instead.

Reply:
Error returned.